devmand/gateway/src/devmand/channels/cli/QueuedCli.cpp

```diff
@@ -44,11 +44,13 @@ QueuedCli::~QueuedCli() {
 }
 
 Future<string> QueuedCli::executeAndRead(const Command& cmd) {
+  boost::recursive_mutex::scoped_lock scoped_lock(mutex);
```

Comment:
why recursive mutex ? instead of ordinary mutex + lock_guard ?

Reply:
because execute calls executeAndRead and should not block on mutex